<launch>
    <param name="deformable_type" value="cloth"/>
    <param name="task_type" value="cloth_table_coverage"/>

    <param name="default_deformability"     value="14.0"/>
    <param name="collision_scaling_factor"  value="1000.0"/>
    <param name="stretching_threshold"      value="$(eval (0.012 + 0.002) / 0.012)"/>
    <param name="max_time"                  value="2.0"/>

    <!-- Table size params -->
    <arg name="table_x_half_extents"    value="0.2"/>
    <arg name="table_y_half_extents"    value="0.2"/>
    <arg name="table_height"            value="0.7"/>
    <arg name="table_thickness"         value="0.05"/>

    <param name="table_x_half_extents"  value="$(arg table_x_half_extents)"/>
    <param name="table_y_half_extents"  value="$(arg table_y_half_extents)"/>
    <param name="table_height"          value="$(arg table_height)"/>
    <param name="table_thickness"       value="$(arg table_thickness)"/>


    <param name="world_x_step" value="0.0125"/>
    <param name="world_x_min" value="-0.3"/>
    <param name="world_x_max" value="0.3"/>

    <param name="world_y_step" value="0.0125"/>
    <param name="world_y_min" value="-0.3"/>
    <param name="world_y_max" value="0.3"/>

    <param name="world_z_step" value="0.0125"/>
    <param name="world_z_min" value="0.5775"/>
    <param name="world_z_max" value="0.7775"/>


    <group ns="smmap_planner_node">
        <param name="visualize_object_desired_motion"   value="false"/>
        <param name="gripper_controller_type"           value="random_sampling"/>
        <param name="max_sampling_counts"               value="5000"/>
        <param name="stretching_cosine_threshold"       value="0.75"/>
    </group>


    <group ns="deform_simulator_node">
        <param name="num_simsteps_per_gripper_command" value="4"/>
    </group>
</launch>
